types/src/address.rs

```diff
@@ -1,14 +1,22 @@
-use casper_types::bytesrepr::{Error, FromBytes};
-use casper_types::{bytesrepr::ToBytes, CLType, CLTyped};
+use crate::{
+    bytesrepr::{Error, FromBytes, ToBytes},
+    CLType, CLTyped,
+};
 
-const ADDRESS_LENGTH: usize = 64;
+/// Max bytes of an [`Address`] internal representation.
+pub const ADDRESS_LENGTH: usize = 64;
 
+/// Blockchain-agnostic address representation.
 #[derive(Clone, Copy, PartialEq, Hash, Eq)]
 pub struct Address {
     data: [u8; ADDRESS_LENGTH],
 }
 
 impl Address {
+    /// Creates a new Address from bytes.
+    ///
+    /// If takes less than [`ADDRESS_LENGTH`], the remaining bytes are zeroed.
+    /// If takes more and [`ADDRESS_LENGTH`] excess bytes are discarded.
```

Comment:
Shouldn't we panic or something instead discarding bytes? It would mean that the backend is not supported by odra or that something went wrong

Reply:
For Casper and MockVM it's works fine. When we have more backends we will have to rethink Address anyway.